aten/src/ATen/TensorBase.h

```diff
@@ -1,16 +1,19 @@
 #pragma once
 
 #include "ATen/TensorImpl.h"
+#include "ATen/UndefinedTensor.h"
 
 namespace at { namespace detail {
 
 // TensorBase is the base class for Tensor which handles the reference counting
 struct TensorBase {
-  TensorBase()
-  : pImpl(nullptr) {}
+  TensorBase(): TensorBase(UndefinedTensor::singleton(), true) {}
   TensorBase(TensorImpl * self, bool retain)
   : pImpl(self) {
-    if(pImpl != nullptr && retain)
+    if (pImpl == nullptr) {
```

Comment:
The move constructor TensorBase(TensorBase && rhs) still assigns `nullptr` to `rhs`. Should that be the undefined tensor? (GitHub won't let me comment at the correct line of code)

Reply:
Yes, good catch.